prog/alloc.go

```diff
@@ -72,7 +76,7 @@ func (ma *memAlloc) alloc(r *randGen, size0 uint64) uint64 {
 		}
 	}
 	ma.bankruptcy()
-	return ma.alloc(r, size0)
+	return ma.alloc(r, size0, alignment)
```

Comment:
This is wrong. We already recalculated alignment in memAllocGranule units, and now we call alloc recursively again, and it will divide it by memAllocGranule again.